test/validation/api/ipsec/ipsec.c

```diff
@@ -1054,7 +1054,7 @@ int ipsec_out_term(void)
 	return ipsec_suite_term();
 }
 
-int ipsec_init(odp_instance_t *inst, odp_ipsec_op_mode_t mode)
+int ipsec_init(odp_instance_t *inst)
```

Comment:
I would have wished for a better commit message for this change. Yes, the mode argument can be removed if we ensure this and that as the message says, but why do we want to do it, what is the purpose of the change? Why is it better to pass mode through suite context instead of the mode argument?

Reply:
I feel that the mode argument is creating an unnecessary copy of information which could be directly inferred from the suite_context if we make sure that suite_context is properly initialized. My thinking was that this change would bring some improvement in the code readability.

Reply:
I am still not quite sure if removing the argument and relying on the global variable (and it being initialized early enough) is an improvement. But I do not feel too strongly about it. Actually, it seems the relevant piece of code gets moved from this function elsewehere in another patch and if the patches are rearranged as I suggest in another commit, then this patch mostly disappears.